src/components/views/EResource.js

```diff
@@ -10,48 +10,47 @@ import Title from './Title';
 import PCI from './PCI';
 import { resourceClasses } from '../../constants';
 
-export default class EResource extends React.Component {
-  static propTypes = {
-    data: PropTypes.shape({
-      eresource: PropTypes.shape({
-        class: PropTypes.string,
-        name: PropTypes.string,
-        type: PropTypes.object,
-      }),
+const propTypes = {
+  data: PropTypes.shape({
+    eresource: PropTypes.shape({
+      class: PropTypes.string,
+      name: PropTypes.string,
+      type: PropTypes.object,
     }),
-    handlers: PropTypes.shape({
-      onClose: PropTypes.func.isRequired,
-      onEdit: PropTypes.func.isRequired,
-      onNeedMorePackageContents: PropTypes.func.isRequired,
-      onToggleTags: PropTypes.func.isRequired
-    }).isRequired,
-    helperApp: PropTypes.func,
-    isLoading: PropTypes.bool,
-  }
+  }),
+  handlers: PropTypes.shape({
+    onClose: PropTypes.func.isRequired,
+    onEdit: PropTypes.func.isRequired,
+    onNeedMorePackageContents: PropTypes.func.isRequired,
+    onToggleTags: PropTypes.func.isRequired
+  }).isRequired,
+  helperApp: PropTypes.func,
+  isLoading: PropTypes.bool,
+};
 
-  renderEditEresourcePaneMenu = () => {
-    const {
-      data: { eresource = {} },
-      handlers,
-    } = this.props;
+const EResource = ({
+  data: { eresource = {} },
+  data = {},
+  helperApp,
+  handlers,
+  isLoading
+}) => {
+  const intl = useIntl();
 
+  const renderEditEresourcePaneMenu = () => {
```

Comment:
Again, use this opportunity to simplify things. You don't need a callback, you don't need a function. Can't you just put the ternary and JSX directly in the `lastMenu` prop?